test/core/credentials/transport/tls/spiffe_utils_test.cc

```diff
@@ -34,106 +34,331 @@
+  ASSERT_TRUE(spiffe_id.ok()) << spiffe_id.status();
+  EXPECT_EQ(spiffe_id->trust_domain(), "example.com");
+  EXPECT_EQ(spiffe_id->path(), "/us");
+}
+
+TEST(SpiffeId, AllSpiffeCapitalizedSuccess) {
+  absl::StatusOr<SpiffeId> spiffe_id =
+      SpiffeId::FromString("SPIFFE://example.com/us");
+  ASSERT_TRUE(spiffe_id.ok()) << spiffe_id.status();
+  EXPECT_EQ(spiffe_id->trust_domain(), "example.com");
+  EXPECT_EQ(spiffe_id->path(), "/us");
+}
+
+TEST(SpiffeId, FirstSpiffeCapitalizedSuccess) {
   absl::StatusOr<SpiffeId> spiffe_id =
-      SpiffeId::FromString(test_case.spiffe_id);
-  ASSERT_TRUE(spiffe_id.ok()) << spiffe_id.status();
-  EXPECT_EQ(spiffe_id->trust_domain(), test_case.trust_domain);
-  EXPECT_EQ(spiffe_id->path(), test_case.path);
-}
-
-INSTANTIATE_TEST_SUITE_P(
-    SpifeIdTestSuccessSuiteInstantiation, SpiffeIdSuccessTest,
-    ::testing::ValuesIn<SpiffeIdSuccessTestCase>({
-        {"spiffe://example.com", "example.com", ""},
-        {"spiffe://example.com/us", "example.com", "/us"},
-        {"sPiffe://example.com/us", "example.com", "/us"},
-        {"SPIFFE://example.com/us", "example.com", "/us"},
-        {"Spiffe://example.com/us", "example.com", "/us"},
-        {"spiffe://example.com/country/us/state/FL/city/Miami", "example.com",
-         "/country/us/state/FL/city/Miami"},
-        {"spiffe://trust-domain-name/path", "trust-domain-name", "/path"},
-        {"spiffe://staging.example.com/payments/mysql", "staging.example.com",
-         "/payments/mysql"},
-        {"spiffe://staging.example.com/payments/web-fe", "staging.example.com",
-         "/payments/web-fe"},
-        {"spiffe://k8s-west.example.com/ns/staging/sa/default",
-         "k8s-west.example.com", "/ns/staging/sa/default"},
-        {"spiffe://example.com/9eebccd2-12bf-40a6-b262-65fe0487d453",
-         "example.com", "/9eebccd2-12bf-40a6-b262-65fe0487d453"},
-        {"spiffe://trustdomain/.a..", "trustdomain", "/.a.."},
-        {"spiffe://trustdomain/...", "trustdomain", "/..."},
-        {"spiffe://trustdomain/abcdefghijklmnopqrstuvwxyz", "trustdomain",
-         "/abcdefghijklmnopqrstuvwxyz"},
-        {"spiffe://trustdomain/abc0123.-_", "trustdomain", "/abc0123.-_"},
-        {"spiffe://trustdomain/0123456789", "trustdomain", "/0123456789"},
-        {"spiffe://trustdomain0123456789/path", "trustdomain0123456789",
-         "/path"},
-    }));
+      SpiffeId::FromString("Spiffe://example.com/us");
+  ASSERT_TRUE(spiffe_id.ok()) << spiffe_id.status();
+  EXPECT_EQ(spiffe_id->trust_domain(), "example.com");
+  EXPECT_EQ(spiffe_id->path(), "/us");
+}
+
+TEST(SpiffeId, LongPathSuccess) {
+  absl::StatusOr<SpiffeId> spiffe_id = SpiffeId::FromString(
+      "spiffe://example.com/country/us/state/FL/city/Miami");
+  ASSERT_TRUE(spiffe_id.ok()) << spiffe_id.status();
+  EXPECT_EQ(spiffe_id->trust_domain(), "example.com");
+  EXPECT_EQ(spiffe_id->path(), "/country/us/state/FL/city/Miami");
+}
+
+TEST(SpiffeId, TrustDomainDashesSuccess) {
```

Comment:
I think you could combine almost all of the rest of these tests into a single one that covers all possible valid characters:

```
TEST(SpiffeId, AcceptedCharacters) {
  auto spiffe_id = SpiffeId::FromString(
      "spiffe://abcdefghijklmnopqrstuvwxyz1234567890.-_/abcdefghijklmnopqrstuvwxyz1234567890.-_");
  ASSERT_TRUE(spiffe_id.ok()) << spiffe_id.status();
  EXPECT_EQ(spiffe_id->trust_domain(), "abcdefghijklmnopqrstuvwxyz1234567890.-_");
  EXPECT_EQ(spiffe_id->path(), "/abcdefghijklmnopqrstuvwxyz1234567890.-_");
}
```